docs/src/main/asciidoc/rest-data-panache.adoc

```diff
@@ -405,6 +406,25 @@ Additionally, you can also filter by the entity fields by adding a query param w
 
 IMPORTANT: Filtering by fields is only supported for primitive types.
 
+== Complex filtering to list entities using @NamedQuery
+
+You can specify a named query to filter when listing the entities. For example, having the following named query in your entity:
+
+[source,java]
+----
+@Entity
+@NamedQuery(name = "Person.containsInName", query = "from Person where name like CONCAT('%', CONCAT(:name, '%'))")
+public class Person extends PanacheEntity {
+  String name;
+}
+----
+
+In this example, we have added a named query to list all the persons that contains some text in the `name` field. 
+
+Next, we can set a query param `namedQuery` when listing the entities using the generated resource with the name of the named query that we want to use, for example, calling `http://localhost:8080/people?namedQuery=#Person.containsInName&name=ter` would return all the persons which name contains the text "ter".
```

Comment:
Is there a good reason why `#` is required for the named query name?

Reply:
It's required by Hibernate to use the named query: https://quarkus.io/guides/hibernate-reactive-panache#named-queries
However, I can add it programmatically.

Reply:
> However, I can add it programmatically.

I think that makes most sense, as I don't see any good reason for a user to be required to add this.

Reply:
Ok, it's done now. PR updated.